src/states/node.rs

```diff
@@ -2032,8 +2032,9 @@ impl Node {
                      dst: Authority<XorName>)
                      -> Result<(), RoutingError> {
         let sections = self.peer_mgr.pub_ids_by_section();
-        let serialised_sections = serialisation::serialise(&sections)?;
-        if digest == sha256::hash(&serialised_sections) {
+        let prefixes = self.peer_mgr.routing_table().prefixes();
+        let serialised_rt = serialisation::serialise(&(&sections, prefixes))?;
```

Comment:
I'm not sure how serialisation works, but won't `&(&sections, prefixes)` and `&(sections, prefixes)` serialise differently? If yes, we might have to clone `sections` here.